core/src/main/scala/kafka/log/Log.scala

```diff
@@ -259,7 +259,7 @@ class Log(@volatile var logStartOffset: Long,
 
   import kafka.log.Log._
 
-  this.logIdent = s"[Log partition=$topicPartition, dir=$parentDir] "
+  this.logIdent = s"[Log partition=$topicPartition, dir=${parentDir}${topicId.map(id => s", topicId=$id").getOrElse("")}] "
```

Comment:
It seems like topicID will almost always be set to the empty string here, since _topicId is (often?) not initialized until these lines below:
```
  locally {
    initializePartitionMetadata()
    updateLogStartOffset(logStartOffset)
    maybeIncrementFirstUnstableOffset()
    initializeTopicId()
  }
```
This might be better in a separate PR?

Reply:
Yeah, fair enough. Will pull this out for now. I'll keep the `toString` change though.